stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/components/selectedmessage/SelectedMessageMenu.kt

```diff
@@ -0,0 +1,137 @@
+package io.getstream.chat.android.compose.ui.components.selectedmessage
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.ColumnScope
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.Reaction
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.common.state.MessageAction
+import io.getstream.chat.android.common.state.React
+import io.getstream.chat.android.compose.handlers.SystemBackPressedHandler
+import io.getstream.chat.android.compose.state.messages.list.MessageOptionState
+import io.getstream.chat.android.compose.ui.components.messageoptions.MessageOptions
+import io.getstream.chat.android.compose.ui.components.messageoptions.defaultMessageOptionsState
+import io.getstream.chat.android.compose.ui.components.reactionoptions.ReactionOptions
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+
+/**
+ * A Composable function that turns [SelectedMessageOptions] into a menu by wrapping it inside of [Surface].
+ *
+ * @param message Used to gather the necessary message information (such as ownership, id, etc.) in order to properly display [SelectedMessageOptions].
+ * @param modifier Compose UI [Modifier] that is applied to the internally used [Surface].
+ * @param shape The [Shape] applied to the internally used [Surface].
+ * @param overlayColor The color applied to the internally used [Box] that wraps [Surface].
```

Comment:
I think we should go over all of these and make them more concise and less hardcoded/specific to the implementation